### PR TITLE
[PM-29113] Check and navigate to vault migration

### PR DESCRIPTION
**Pull request by andrebispo5**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-29113

## 📔 Objective

This pull request introduces functionality to prompt users to migrate their personal vault items to an organization when certain policy and feature flag conditions are met. It adds a new method to determine the earliest organization applying a specific policy, updates the `Policy` model to include a revision date, and integrates these changes into the sync process. Comprehensive tests are included for the new logic.

**Vault migration feature integration:**

* Added a private method `checkUserNeedsVaultMigration()` to `DefaultSyncService` that checks if the user needs to migrate personal vault items to an organization based on a feature flag, the presence of the Personal Ownership policy, and the existence of personal vault items. If all conditions are met, it notifies the delegate via `migrateVaultToMyItems(organizationId:)`. 
* Updated the `SyncServiceDelegate` protocol to include the `migrateVaultToMyItems(organizationId:)` method, which is called when a migration is required.

**Policy service enhancements:**

* Added the `getEarliestOrganizationApplyingPolicy(_:)` method to `PolicyService` and implemented it in `DefaultPolicyService`. This method returns the organization ID with the earliest revision date for a given policy type.
* Updated the `Policy` model to include an optional `revisionDate` property and ensured it is populated from response models and fixtures. 
* Updated policy fixtures to support the new `revisionDate` property for testing purposes.

**Dependency and test updates:**

* Added `configService` as a dependency to `ServiceContainer`, `DefaultSyncService`, and related test setups to support feature flag checks. 
* Added comprehensive tests for the new `getEarliestOrganizationApplyingPolicy(_:)` logic, covering edge cases such as no policies, multiple organizations, nil revision dates, and filtering by policy type.

## 📸 Screenshots

https://github.com/user-attachments/assets/c246b94d-e71c-4d26-8231-b06804de2a68

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: BitwardenShared/Core/Vault/Services/PolicyService.swift [hunk 1]
```diff
@@ -373,6 +380,18 @@ extension DefaultPolicyService {
         return [.card]
     }
 
+    func getEarliestOrganizationApplyingPolicy(_ policyType: PolicyType) async -> String? {
+        let policies = await policiesApplyingToUser(policyType, filter: nil)
+        guard !policies.isEmpty else { return nil }
```
Comment (fedemkr) [inline BitwardenShared/Core/Vault/Services/PolicyService.swift]: ⛏️ `min(by:)` already returns `nil` if there are no elements so we can skip this guard if you want ([docs](https://developer.apple.com/documentation/swift/array/min(by:)#return-value)).

File: BitwardenShared/Core/Vault/Services/PolicyServiceTests.swift [hunk 2]
```diff
@@ -336,6 +336,136 @@ class PolicyServiceTests: BitwardenTestCase { // swiftlint:disable:this type_bod
...
+            ),
+        ])
+
+        let organizationId = await subject.getEarliestOrganizationApplyingPolicy(.personalOwnership)
+
+        XCTAssertEqual(organizationId, "org-2")
+    }
+
+    /// `getEarliestOrganizationApplyingPolicy()` handles policies with nil revision dates.
+    func test_getEarliestOrganizationApplyingPolicy_nilRevisionDates() async {
```
Comment (fedemkr) [inline BitwardenShared/Core/Vault/Services/PolicyServiceTests.swift:405]: 🤔 Could you add a test where all policies have `revisionDate` as `nil`? Is the expected behavior here correct on the logic when all are `nil`?

File: BitwardenShared/Core/Vault/Services/SyncService.swift [hunk 3]
```diff
@@ -497,6 +511,32 @@ extension DefaultSyncService {
...
+    ///
+    /// The user needs to migrate if:
+    /// - The feature flag is enabled
+    /// - The user is a member of an organization with the Personal Ownership policy enabled
+    /// - The user has one or more items in their personal vault (including deleted items)
+    ///
+    private func checkUserNeedsVaultMigration() async throws {
+        // Check if feature flag is enabled
+        guard await configService.getFeatureFlag(.migrateMyVaultToMyItems) else { return }
+        // Get the earliest organization applying the personal ownership policy
```
Comment (fedemkr) [inline BitwardenShared/Core/Vault/Services/SyncService.swift]: ⛏️ I believe the code is self-explanatory here and the comments can be removed.

File: BitwardenShared/Core/Vault/Services/SyncService.swift [hunk 4]
```diff
@@ -497,6 +511,32 @@ extension DefaultSyncService {
...
     }
 
+    /// Checks if the user needs to migrate their personal vault items to an organization.
+    ///
+    /// The user needs to migrate if:
+    /// - The feature flag is enabled
+    /// - The user is a member of an organization with the Personal Ownership policy enabled
+    /// - The user has one or more items in their personal vault (including deleted items)
+    ///
+    private func checkUserNeedsVaultMigration() async throws {
```
Comment (fedemkr) [inline BitwardenShared/Core/Vault/Services/SyncService.swift:528]: 🎨 Instead of doing this logic for the app extensions and then checking on the delegate call whether it's on an app extension to not continue, you can do it directly here by injecting `AppContextHelper` and then you can directly check it without needing the appExtensionDelegate:
```swift
guard appContextHelper.appContext == .mainApp else {
    return
}
```
This way we can save resources and not fetch all ciphers into memory on extensions.

File: BitwardenShared/UI/Platform/Application/AppCoordinator.swift [hunk 5]
```diff
@@ -276,6 +279,32 @@ class AppCoordinator: Coordinator, HasRootNavigator {
         }
     }
 
+    /// Show the migrate to my items screen.
+    ///
+    /// - Parameter organizationId: The organization ID that requires the vault migration.
+    ///
+    private func showMigrateToMyItems(organizationId: String) {
+        DispatchQueue.main.async {
```
Comment (fedemkr) [inline BitwardenShared/UI/Platform/Application/AppCoordinator.swift]: 🤔 Is `DispatchQueue.main.async` necessary here? I think this is already being called from the main thread given the call site:
```swift
@MainActor
func migrateVaultToMyItems(organizationId: String)
```

File: BitwardenShared/UI/Platform/Application/AppProcessor.swift [hunk 6]
```diff
@@ -655,6 +655,14 @@ extension AppProcessor: SyncServiceDelegate {
             coordinator?.navigate(to: .auth(.setMasterPassword(organizationIdentifier: orgIdentifier)))
         }
     }
+
+    func migrateVaultToMyItems(organizationId: String) {
+        // Don't show the migrate vault screen if running in an app extension.
+        guard appExtensionDelegate?.isInAppExtension != true else { return }
+
+        coordinator?.hideLoadingOverlay()
```
Comment (fedemkr) [inline BitwardenShared/UI/Platform/Application/AppProcessor.swift]: 🤔 Is the coordinator showing a loading overlay on the extensions as well? If so, it won't be hidden as the guard will return before `coordinator?.hideLoadingOverlay()` line.

**Review by fedemkr — COMMENTED** [review]
Great job Andre! Just a few comments.
